sorts/quick_sort.ts

```diff
@@ -10,29 +10,39 @@ export const partition = (
   left: number = 0,
   right: number = array.length - 1
 ) => {
-  const pivot = array[Math.floor((right + left) / 2)];
-  let i = left;
+  const pivotIndex = choosePivot(left,right);
+  const pivot = array[pivotIndex];
+  [array[pivotIndex], array[right]] = [array[right], array[pivotIndex]];
+  let i = left - 1;
   let j = right;
 
-  while (i <= j) {
-    while (array[i] < pivot) {
-      i++;
-    }
-
-    while (array[j] > pivot) {
-      j--;
-    }
+  while (i < j) {
+    while (array[++i] < pivot);
+    while (array[--j] > pivot);
 
-    if (i <= j) {
+    if (i < j) {
       [array[i], array[j]] = [array[j], array[i]];
-      i++;
-      j--;
```

Comment:
Because this is conditionally done, we might at times overshoot the index at which the comparison transition with `pivot` element happens. To fix this, I've (more or less) moved this increment inside the inner `while` loops.